autoload/go/cmd.vim

```diff
@@ -117,16 +117,16 @@ endfunction
 
 " Run runs the current file (and their dependencies if any) in a new terminal.
 function! go#cmd#RunTerm(bang, mode, files) abort
-  let cmd = "go run "
+  let cmd = ["go", "run"]
   let tags = go#config#BuildTags()
-  if len(tags) > 0
-    let cmd .= "-tags " . go#util#Shellescape(tags) . " "
+  if go#config#BuildTags() > 0
+    call extend(cmd, ["-tags", go#util#Shellescape(tags)])
   endif
 
   if empty(a:files)
-    let cmd .= go#util#Shelljoin(go#tool#Files())
+    call add(cmd, go#util#Shelljoin(go#tool#Files()))
```

Comment:
Okay @bhcleek 
This Shelljoin call is still breaking the term_start function. the cmd var looks like this after this method runs:
```
['go', 'run', '''/home/louis/git/go/claircore/cmd/libindexhttp/main.go''']
```
When I swap the Shelljoin method with just "join" I get a correct arg list

```
['go', 'run', '/home/louis/git/go/claircore/cmd/libindexhttp/main.go']
```

may need a little help here.

Reply:
You're on the right track. Using `join` instead of `go#util#Shelljoin` would be the right thing if the list of files were a single argument, but each file is an argument. I think what's needed here is
```
call exend(cmd, go#tool#Files())
```

Reply:
worked :+1: